utils/crc.hh

```diff
@@ -105,12 +105,12 @@ public:
             --size;
         }
         if ((reinterpret_cast<uintptr_t>(in) & 3) && size >= 2) {
-            process_le(*reinterpret_cast<const uint16_t*>(in));
+            process_le(seastar::read_le<uint16_t>(reinterpret_cast<const char*>(in)));
```

Comment:
Hmm, why is this reinterpret_cast fine?

Reply:
And a second question - don't read_le and process_le both do byte swapping - so it's wrong?
Of course this question applies to many other places below as well.

Reply:
`read_le` only swaps bytes on big-endian architectures. This particular piece of code is only used on little-endian machines (`#if defined(__x86_64__) || defined(__i386__) || defined(__aarch64__)`), so `read_le` does the same thing as the `reinterpret_cast`, but correctly.